app/controllers/locations_controller.rb

```diff
@@ -88,6 +88,7 @@ def show
     request.query_parameters["layout"] = true
     @query_parameters = request.query_parameters
     @url = request.url
+    @url_params = @url.split('?').last
   end
```

Comment:
I see on line 89 that the `@query_parameters` are being fetched from the request

Those can be used on the view as follows: 
`= link_to 'Community Resources', locations_path(@query_parameters)`

This will be a cleaner way than performing the string interpolation.